modules/puppetca_http_api/puppetca_impl.rb

```diff
@@ -0,0 +1,31 @@
+module ::Proxy::PuppetCa::PuppetcaHttpApi
+  class PuppetcaImpl
+    extend Proxy::PuppetCa::DependencyInjection
+    inject_attr :http_api_impl, :client
+
+    def sign(certname)
+      client.sign(certname)
+    end
+
+    def clean(certname)
+      client.clean(certname)
+    end
+
+    # list of all certificates and their state/fingerprint
+    def list
+      response = client.search
+      response.each_with_object({}) do |entry, hsh|
+        name = entry['name']
+        # serial, not_before and not_after are not available via http api
```

Comment:
Looks like they will be. Should we already start reading them anyway so we'll support it when it is? Should return nil if it's not found either way.

Reply:
Let's wait until a version is released that contains this data.

Reply:
It was just merged so I think we can start assuming the field names.

Reply:
I won't add any new features to this PR. Let's merge this first.
We have to make sure our code can handle the old and new versions, so we need appropriate tests etc. I'd prefer to do this later. Keep in mind this PR was opened in October, we have user testing, unit tests, code reviews. What is keeping us from merging?